pkg/apis/kops/v1alpha2/cluster.go

```diff
@@ -637,3 +640,25 @@ type RollingUpdate struct {
 	// +optional
 	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
 }
+
+// AzureSpec defines Azure specific cluster configuration.
+type AzureSpec struct {
+	// SubscriptionID specifies the subscription used for the cluster installation.
+	SubscriptionID string `json:"subscriptionId,omitempty"`
+	// TenantID is the ID of the tenant that the cluster is deployed in.
+	TenantID string `json:"tenantId"`
+	// ResourceGroupName specifies the name of the resource group
+	// where the cluster is built.
+	// If this is empty, kops will create a new resource group
+	// whose name is same as the cluster name. If this is not
+	// empty, kops will not create a new resource group, and
+	// it will just reuse the existing resource group of the name.
+	// This follows the model that kops takes for AWS VPC.
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
Optional API fields typically use pointers to more easily denote nil as being unset.

Reply:
I don't agree that is a necessary pattern. Using a pointer is only necessary when it is useful to distinguish the default value from an explicitly specified zero value. In this case an exiting resource group named with the empty string would not be reasonable.